### PR TITLE
Fb831

### PR DESCRIPTION
**Pull request by jiminssyspire**
There are some code left unfinished. But this pull request has completed base ViewModel and DefaultStyleRule.

### CODE REVIEW

**Comment by jiminssyspire** [comment]
I realized that default style rule only needs to be instantiated once.

**Comment by jiminssyspire** [comment]
code review changes are done.

**Comment by jiminssyspire** [comment]
I put in tests for the transforms as well.

**Comment by jiminssyspire** [comment]
yeah, simplified and sliced the array.

File: Applications/ClientWebPortal/Source/webapp/spec/src/utils/table-models/view-model/spec.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        expect(payload.original[2].display).toBe('Will Smith');
+        expect(payload.original[2].style instanceof DefaultStyleRule).toBe(true);
+        done();
+      });
+      sourceModel.emitDataUpdateSignal(1);
+    });
+  });
+});
+
+class MockSourceModel {
```
Comment (kranar) [inline Applications/ClientWebPortal/Source/webapp/spec/src/utils/table-models/view-model/spec.js]: Use ArrayModel for testing, don't create a new class for it as this class may contain bugs.

Comment (jiminssyspire) [reply Applications/ClientWebPortal/Source/webapp/spec/src/utils/table-models/view-model/spec.js]: Done.

**Pull request merged**